### PR TITLE
First stab at base class + events helper

### PR DESCRIPTION
**Pull request by LeaVerou**

### CODE REVIEW

File: src/common/Element.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			this.initializeProps?.();
+
+			this.#initialized = true;
+		}
+	}
+
+
+	static init () {
+		// One time stuff
+		if (this._initialized) {
```
Comment (DmitrySharabin) [inline src/common/Element.js:33]: Is it some sort of static variable that shouldn't be defined? I don't understand how we can use uninitialized variables in static methods. I didn't find it `_initialized` in other parts of the code base. Does the `HTMLElement` class (instance) have it?

File: src/common/PropChangeEvent.js [hunk 2]
```diff
@@ -7,52 +7,41 @@ export class PropChangeEvent extends CustomEvent {
...
+		Object.defineProperty(CustomPropChangeEvent, "name", { value: eventClassName });
+		return CustomPropChangeEvent;
+	}
+}
 
-			this[name] = options[name];
-		}
-	};
-	Object.defineProperty(CustomPropChangeEvent, "name", { value: eventClassName });
+export default function definePropChangeEvent (Class, name) {
```
Comment (DmitrySharabin) [inline src/common/PropChangeEvent.js:36]: Looking at the implementation, I can't see why we need the `Class` parameter—it's not used anywhere. Does this helper can be used without a class at all?

Comment (LeaVerou) [reply src/common/PropChangeEvent.js]: Good point, removed!

File: src/common/Element.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+	}
+
+	connectedCallback () {
+		if (!this.#initialized) {
+			this.initializeProps?.();
+
+			this.#initialized = true;
+		}
+	}
```
Comment (DmitrySharabin) [inline src/common/Element.js:28]: For some reason, this code is not reachable. Because of this, `<color-slider>` is broken.

Comment (DmitrySharabin) [reply src/common/Element.js:28]: Should we call `super.connectedCallback()` from the subclasses?

Comment (DmitrySharabin) [reply src/common/Element.js:28]: Is it legal at all? :)

Comment (LeaVerou) [reply src/common/Element.js:28]: > Should we call `super.connectedCallback()` from the subclasses?

Oh yes, absolutely. I must have forgotten

File: src/common/Element.js (hunk 1 above)
Comment (DmitrySharabin) [reply src/common/Element.js:33]: Got it. Thanks!

**Pull request merged**